File: common/rikmuld/camping/item/ItemLantern.java
package rikmuld.camping.item;

import java.util.List;

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;

public class ItemLantern extends ItemMain {

	public ItemLantern(String name)
	{
		super(name);
		this.setHasSubtypes(true);
		this.setMaxDamage(600);
		this.setMaxStackSize(1);
		this.canRepair = false;
	}
	
    public void addInformation(ItemStack stack, EntityPlayer player, List list, boolean par4)
    {
    	list.add("Burning time left: "+(stack.getMaxDamage()-stack.getItemDamage())/2+" seconds");
    }
}

File: scala/com/rikmuld/camping/features/blocks/lantern/ItemLantern.scala
package com.rikmuld.camping.features.blocks.lantern

import com.rikmuld.camping.Definitions.Lantern
import com.rikmuld.corerm.objs.ObjDefinition
import com.rikmuld.corerm.objs.items.ItemBlockRM
import net.minecraft.block.Block
import net.minecraft.client.util.ITooltipFlag
import net.minecraft.item.ItemStack
import net.minecraft.world.World

class ItemLantern(modId: String, definition: ObjDefinition, block: Block)
  extends ItemBlockRM(modId, definition, block) {

    override def addInformation(stack: ItemStack,
                                player: World,
                                list: java.util.List[String],
                                tooltip: ITooltipFlag): Unit =
      if (stack.hasTagCompound)
        list.add("Burning time left: " +
          stack.getTagCompound.getInteger("time") + " seconds")
      else if(stack.getItemDamage == Lantern.ON)
        list.add("Burning time left: 750 seconds")
  }